src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -214,6 +223,12 @@ void ErrorHandler(object sender, Newtonsoft.Json.Serialization.ErrorEventArgs ev
                         }
                     }
 
+                    if (!(exception is JsonException || exception is OverflowException))
+                    {
+                        var exceptionDispatchInfo = ExceptionDispatchInfo.Capture(exception);
+                        exceptionDispatchInfo.Throw();
+                    }
+
```

Comment:
This only chance to this class should be the new property/interface. You aren't required to throw exceptions when you have invalid input, and it's non-ideal.